package/doc/sphinx/source/documentation_pages/coordinates/pickle_readers.rst

```diff
@@ -0,0 +1,80 @@
+.. Contains the formatted docstrings for the serialization of universe located 
+.. mainly in 'MDAnalysis/libs/pickle_file_io.py'
+.. _serialization:
+
+*********************************************************
+Serialization of Coordinate Readers
+*********************************************************
+
+To achieve a flawless implementation of parallelism, this document illustrates
+the basic idea of how different coordinate readers are being serialized in MDAnalysis,
+and what developers should do to serialize a new reader.
+
+To make sure every Trajectory reader can be successfully
+serialized, we implement picklable I/O classes (see :ref:`implemented-fileio`).
+When the file is pickled, filename and other necessary attributes of the open 
+file handle are saved. On unpickling, the file is opened by filename.
+This means that for a successful unpickle, the original file still has to
+be accessible with its filename. To retain the current frame of the trajectory,
+:func:`_read_frame(previous frame)` will be called during unpickling.
+
+.. _how_to_serialize_a_new_reader:
+
+How to serialize a new reader
+-----------------------------
+
+File Access
+^^^^^^^^^^^
+If the new reader uses :func:`util.anyopen()` (e.g. PDB), the reading handler
+can be pickled without modification.
+If the new reader uses I/O classes from other package (e.g. GSD), and cannot
+be pickled natively, create a new picklable class inherited from 
+the file class in that package (e.g. GSDPicklable), adding :func:`__getstate__`,
+:func:`__setstate__` functions (or :func:`__reduce__` if needed. Consult the
+pickle [documentation](https://docs.python.org/3/library/pickle.html) of python)
+to allow file handler serialization.
+
+To seek or not to seek
+^^^^^^^^^^^^^^^^^^^^^^
+Some I/O class supports :func:`seek` and :func:`tell` functions to allow the file 
+to be pickled with an offset. It is normally not needed for MDAnalysis with
+random access. But if error occurs during testing, find a way to make the offset work.
+Maybe this I/O class supports frame indexing? Maybe the file handler inside this I/O 
+class supports offset?
```

Comment:
Can you give example of the cases. This will make it easier for others to understand.